dashboard-app/src/hooks/useErrorsEffect.ts

```diff
@@ -0,0 +1,21 @@
+import { Dispatch, SetStateAction, useEffect } from 'react';
+import { AxiosError } from 'axios';
+import { AggregatedData } from '../dashboard/dataManipulation/logsToAggregatedData';
+import { Dictionary } from 'ramda';
+
+export const useErrorsEffect = (
```

Comment:
There is some core semantic being expressed here that has nothing particularly to do with either errors or aggregate data.

It feels like something like this, which is something I was trying to grasp at in my note in #184:
```ts
const useResettableState = <T>(init: T, updateOn: any[], resetOn: any[]) => {
	// ...
};

// Usage
const [errors, setErrors] = useResettableState({}, [error], [data]);
```

Reply:
To get this working with the current implementation, we would also need to supply a callback to describe how we are updating state. I think it would be simpler if we removed `onUpdate`, and handled this separately:

```js
const useResettableState = <T>(init: T, resetOn: any[]) => {
	// ...
};
const [errors, setErrors] = useResettableState({}, [data]);
```

Reply:
Yeah good point. I suspect if I had tried to fill out `// ...` in my example I'd have realised that, lol.

However, we could still maybe keep your error hook, and just compose it with a `useResettableState` hook. 

So in the end this might look like:
```ts
const useAxiosErrors(resetOn: any[]) => {
  const [errors, setErrors] = useResettableState<Dictionary<string>>({}, resetOn);

  const setState = useCallback((error: AxiosError, key: string) => {
    setErrors(assoc(key, error.message))
  }, []);

  return [errors, setState];
}
```